src/hotspot/os_cpu/bsd_aarch64/vm_version_bsd_aarch64.cpp

```diff
@@ -188,29 +188,38 @@ unsigned long VM_Version::os_get_processor_features() {
   id_aa64pfr0 = READ_SPECIALREG(ID_AA64PFR0_EL1);
 
   if (ID_AA64ISAR0_AES(id_aa64isar0) == ID_AA64ISAR0_AES_BASE) {
-    auxv = auxv | HWCAP_AES;
+    auxv = auxv | CPU_AES;
```

Comment:
Since the code that uses the auxv value returned from this method uses HWCAP_* defines to inspect it, I think we should not move to internal JDK defines that happen to have the same values.

I would think something like this at the top of this file would be sufficient:
```
#ifndef HWCAP_ASIMD
#define HWCAP_ASIMD (1<<1)
#endif

#ifndef HWCAP_AES
#define HWCAP_AES   (1<<3)
#endif

#ifndef HWCAP_PMULL
#define HWCAP_PMULL (1<<4)
#endif

#ifndef HWCAP_SHA1
#define HWCAP_SHA1  (1<<5)
#endif

#ifndef HWCAP_SHA2
#define HWCAP_SHA2  (1<<6)
#endif

#ifndef HWCAP_CRC32
#define HWCAP_CRC32 (1<<7)
#endif

#ifndef HWCAP_ATOMICS
#define HWCAP_ATOMICS (1<<8)
#endif

#ifndef ID_AA64PFR0_AdvSIMD_SHIFT
#define ID_AA64PFR0_AdvSIMD_SHIFT 20
#endif

#ifndef ID_AA64PFR0_AdvSIMD
#define ID_AA64PFR0_AdvSIMD(x) ((x) & (UL(0xf) << ID_AA64PFR0_AdvSIMD_SHIFT))
#endif

#ifndef ID_AA64PFR0_AdvSIMD_IMPL
#define ID_AA64PFR0_AdvSIMD_IMPL (UL(0x0) << ID_AA64PFR0_AdvSIMD_SHIFT)
#endif

#ifndef ID_AA64PFR0_AdvSIMD_HP
#define ID_AA64PFR0_AdvSIMD_HP (UL(0x1) << ID_AA64PFR0_AdvSIMD_SHIFT)
#endif
```

Reply:
<<...I think we should not move to internal JDK defines that happen to have the same values.>>
that was my first thought but I liked the word "CPU" :-), so  I understand your claim for HWCAP : .. #ifndef HWCAP .... I will change that..

<<... #ifndef ID_AA64PFR0_AdvSIMD.....>>
since for FreeBSD > 12.0 these defines are here : 
https://svnweb.freebsd.org/base/head/sys/arm64/include/armreg.h?revision=350242&view=co 
.. do we really need to #ifndef them  in this file?
the problem is that for FreeBSD-versions < 12.1 these defines are called e.g. `ID_AA64PFR0_ADV_SIMD_SHIFT`  instead of `ID_AA64PFR0_AdvSIMD_SHIFT`  : 
https://svnweb.freebsd.org/base/release/12.0.0/sys/arm64/include/armreg.h?revision=341707&view=markup
that's the reason for  : 
```
/* FreeBSD < 12.1 */
  #if defined(ID_AA64PFR0_ADV_SIMD)
```
```
/* FreeBSD > 12.0 */
	#elif defined(ID_AA64PFR0_AdvSIMD)
```
at the end of the file . that solved the compiling issues for fbsd 12.0